src/pjpersist/tests/test_datamanager.py

```diff
@@ -1612,10 +1613,74 @@ def test_dirty_dm_tpc_prepare(self):
         self.assertTrue('tpc_prepare' in str(prep_mock_p.call_args_list))
         self.assertTrue('tpc_commit' in str(prep_mock_p.call_args_list))
 
+    def test_release_on_destroy(self):
+        # When DM is destroyed without committing or aborting the transaction,
+        # connection should be released.
+        # GIVEN
+        conn = testing.getConnection(testing.DBNAME)
+        pool = testing.DummyConnectionPool(conn)
+        dm = datamanager.PJDataManager(pool)
+        dm.begin()
+        self.assertTrue(pool.isTaken())
+
+        # WHEN
+
+        # Create new transaction to release the reference to our data manager
+        # and allow it to be destroyed
+        transaction.begin()
```

Comment:
I wonder how this will play with the garbage collector

Reply:
Why does it have to play with garbage collector?

Reply:
`__del__` will be called by the garbage collector, when it destroys the object right? I think unless you call `gc.collect` that can happend at at 'random' times

Reply:
`__del__` is there as a safeguard. It is not expected to be ever executed. But if something goes awfully wrong, we are not going to be piling up db taken connection in the pool.

Reply:
Added a test that explicitly tests `__del__` (`test_release_on_destroy`)